python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -150,6 +160,10 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
             out=out,
             required_resource_keys=self.required_resource_keys,
             tags={"kind": self.compute_kind} if self.compute_kind else None,
+            config_schema={
+                "input_asset_partitions": Field(dict, is_required=False),
+                "output_asset_partitions": Field(dict, is_required=False),
+            },
```

Comment:
do we not expect `Asset` to ever be able to define config schema?

Reply:
I was just chatting about this with @smackesey. I think we probably do ultimately.  At least initially, I'd probably want to make it mutually exclusive with asset partitions. Making an asset configurable is essentially like giving it a set of partitions that's determinable at runtime, so combining that with definition-time partitions would be quite gnarly.

One option would be to have some reserved config keys and merge the user-provided config schema with this config schema?